commit d6b9a81b2a45786384f5bd3516bd6ddfb4b772c6
Author: Anton Blanchard <anton@samba.org>
Date:   Sun Jun 24 18:26:17 2012 +0000

    powerpc: IOMMU fault injection
    
    Add the ability to inject IOMMU faults. We enable this per device
    via a fail_iommu sysfs property, similar to fault injection on other
    subsystems.
    
    An example:
    
    ...
    0003:01:00.1 Ethernet controller: Emulex Corporation OneConnect 10Gb NIC (be3) (rev 02)
    
    To inject one error to this device:
    
    echo 1 > /sys/bus/pci/devices/0003:01:00.1/fail_iommu
    echo 1 > /sys/kernel/debug/fail_iommu/probability
    echo 1 > /sys/kernel/debug/fail_iommu/times
    
    As feared, the first failure injected on the be3 results in an
    unrecoverable error, taking down both functions of the card
    permanently:
    
    be2net 0003:01:00.1: Unrecoverable error in the card
    
    Signed-off-by: Anton Blanchard <anton@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

diff --git a/arch/powerpc/Kconfig.debug b/arch/powerpc/Kconfig.debug
index e5f26890a69e..5416e28a7538 100644
--- a/arch/powerpc/Kconfig.debug
+++ b/arch/powerpc/Kconfig.debug
@@ -331,4 +331,13 @@ config STRICT_DEVMEM
 
 	  If you are unsure, say Y.
 
+config FAIL_IOMMU
+	bool "Fault-injection capability for IOMMU"
+	depends on FAULT_INJECTION
+	help
+	  Provide fault-injection capability for IOMMU. Each device can
+	  be selectively enabled via the fail_iommu property.
+
+	  If you are unsure, say N.
+
 endmenu